dev/com.ibm.ws.classloading/src/com/ibm/ws/classloading/internal/ClassLoaderRef.java

```diff
@@ -0,0 +1,160 @@
+/*******************************************************************************
+ * Copyright (c) 2025 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package com.ibm.ws.classloading.internal;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.instrument.ClassFileTransformer;
+import java.lang.ref.WeakReference;
+import java.net.URL;
+import java.security.ProtectionDomain;
+import java.util.EnumSet;
+import java.util.Enumeration;
+import java.util.Objects;
+
+import org.osgi.framework.Bundle;
+
+import com.ibm.websphere.ras.annotation.Trivial;
+import com.ibm.wsspi.classloading.ApiType;
+
+/**
+ * This class is used to create either weak references to AppClassLoaders
+ */
+@Trivial
+public class ClassLoaderRef extends LibertyLoader implements SpringLoader {
+    static {
+        ClassLoader.registerAsParallelCapable();
+    }
+
+    private final WeakReference<AppClassLoader> classLoaderWeakRef;
+
+    ClassLoaderRef(AppClassLoader appClassLoader) {
+        super(appClassLoader.getParent());
```

Comment:
Will this still pin the gateway class loader?

Reply:
Yes it will, and I figured that was a good thing so that the appropriate APIs and SPIs are available to the thread context classloader still.  I would expect that to be a breaking change if threads no longer were able to access those.